bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/ModelController.java

```diff
@@ -39,8 +42,31 @@
 
    void getValidationConstraints(Context ctx, String modeluri);
 
+   /**
+    * V1 version of the executeCommand API. This method expects
+    * a {@link CCommand} payload, serialized as Json or XMI (Or
+    * any application-specific format).
+    *
+    * @param ctx
+    * @param modelURI
+    */
    void executeCommand(Context ctx, String modelURI);
 
+   /**
+    * V2 version of the executeCommand API. This methods expects
+    * a Json Object with a "type" attribute, where type may be
+    * either {@link ModelServerPathsV2#JSON_PATCH} or
+    * {@link ModelServerPathsV2#EMF_COMMAND}, and a "data" attribute
+    * representing a Command or a Patch, in a format that corresponds
+    * to the specified type.
+    *
+    * @param ctx
+    * @param modelURI
+    */
+   default void executeCommandV2(final Context ctx, final String modelURI) {
```

Comment:
why define it as default and throw an exception?

Reply:
The idea is that current custom V1 implementations won't have to change anything (but they won't get any support for V2 either). So they can keep running without any change, and if they want to support V2, they'll have to implement the new methods.

If clients try to reach the V2 API for these V1 implementations, they'll receive the exception.